lib/secure_headers/cookie.rb

```diff
@@ -0,0 +1,78 @@
+require 'cgi'
+
+module SecureHeaders
+  class Cookie
+    SECURE_REGEXP = /;\s*secure\s*(;|$)/i.freeze
+    HTTPONLY_REGEXP =/;\s*HttpOnly\s*(;|$)/i.freeze
+    SAMESITE_REGEXP =/;\s*SameSite\s*(;|$)/i.freeze
+
+    REGEXES = {
+      secure: SECURE_REGEXP,
+      httponly: HTTPONLY_REGEXP,
+      samesite: SAMESITE_REGEXP,
+    }
+
+    attr_reader :raw_cookie, :config
+
+    def initialize(cookie, config)
+      @raw_cookie = cookie
+      @config = config
+    end
+
+    def to_s
+      @raw_cookie.dup.tap do |c|
+        c << "; secure" if secure?
+        c << "; HttpOnly" if httponly?
+        c << "; SameSite" if samesite?
+      end
+    end
+
+    def secure?
+      already_flagged?(:secure) || flag_cookie?(:secure)
```

Comment:
I think this would be better if the conditions were swapped so we can short circuit on a hash lookup instead of a regex match.